lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -314,15 +314,53 @@ static bool allIndexingsAreProjectedPermutation(linalg::GenericOp genericOp) {
   });
 }
 
-// Return true if the linalg.generic can be mapped to a tpp.add.
-bool isTppAdd(linalg::GenericOp linalgOp) {
+// Only parallel loops at buffer semantics with static shapes.
+bool hasMappingToTppConditions(linalg::GenericOp linalgOp) {
   if (linalgOp.getNumLoops() != linalgOp.getNumParallelLoops())
     return false;
-  if (linalgOp.getNumDpsInits() != 1)
-    return false;
   if (linalgOp.hasTensorSemantics())
     return false;
-  if (!hasStaticShape(linalgOp))
+  return hasStaticShape(linalgOp);
+}
+
+// Return true if the operation is binary.
+bool isBinaryOp(linalg::GenericOp linalgOp) {
+  if ((linalgOp.getNumDpsInputs() == 1) && (linalgOp.getNumDpsInits() == 1)) {
```

Comment:
The uses are the one in the region of the generic. Just want to make sure we are using the input/output instead of other values, because the lianlg.generic is non-isolated from above and can capture outside IR.

Reply:
Makes sense.